src/Security/Voter/ReportSystemVoter.php

```diff
@@ -6,42 +6,25 @@
 use App\Entity\System;
 use App\Entity\User;
 use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
-use Symfony\Component\Security\Core\Authorization\Voter\Voter;
 
-class ReportSystemVoter extends Voter
+class ReportSystemVoter extends AbstractVoter
 {
-    const SHOW = 'show';
-    const EDIT = 'edit';
-
     protected function supports($attribute, $subject)
     {
-        if (!in_array($attribute, [self::SHOW, self::EDIT])) {
-            return false;
-        }
-
-        if (!$subject instanceof Report && !$subject instanceof System) {
-            return false;
-        }
-
-        return true;
+        return $this->isSupported($attribute, $subject, [Report::class, System::class]);
     }
 
     protected function voteOnAttribute($attribute, $subject, TokenInterface $token)
     {
-        $user = $token->getUser();
-        // if the user is anonymous, do not grant access
-        if (!$user instanceof User) {
-            return false;
-        }
-
-        // Admins can access all.
-        if ($user->hasRole('ROLE_ADMIN') || $user->hasRole('ROLE_SUPER_ADMIN')) {
+        if ($this->userHasAccessDirectAccess($subject, $token)) {
             return true;
         }
 
+        $user = $token->getUser();
         $userGroups = array_map(function($e) {
             return is_object($e) ? $e->getId() : null;
         }, $user->getGroups()->toArray());
+
```

Comment:
`array_map` is not added by this PR, but `$user->getGroups()->map` should be used here.